src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -514,32 +513,21 @@ private function getNamespacePrefixes()
      */
     public function getNamespaces()
     {
-        return (array) $this->getNamespacesObject();
-    }
+        if (!isset($this->namespaces)) {
+            $query = 'SELECT prefix, uri FROM phpcr_namespaces';
+            $result = $this->getConnection()->query($query);
+            $namespaces = $result->fetchAll(\PDO::FETCH_KEY_PAIR);
 
-    /**
-     * Return the namespaces of the current session as a referenceable ArrayObject.
-     *
-     * @return \ArrayObject
-     */
-    protected function getNamespacesObject()
-    {
-        if ($this->namespaces->count() === 0) {
-            $query = 'SELECT * FROM phpcr_namespaces';
-            $data = $this->getConnection()->fetchAll($query);
-
-            $this->namespaces->exchangeArray(array(
+            $this->namespaces = array(
                 NamespaceRegistryInterface::PREFIX_EMPTY => NamespaceRegistryInterface::NAMESPACE_EMPTY,
                 NamespaceRegistryInterface::PREFIX_JCR => NamespaceRegistryInterface::NAMESPACE_JCR,
                 NamespaceRegistryInterface::PREFIX_NT => NamespaceRegistryInterface::NAMESPACE_NT,
                 NamespaceRegistryInterface::PREFIX_MIX => NamespaceRegistryInterface::NAMESPACE_MIX,
                 NamespaceRegistryInterface::PREFIX_XML => NamespaceRegistryInterface::NAMESPACE_XML,
                 NamespaceRegistryInterface::PREFIX_SV => NamespaceRegistryInterface::NAMESPACE_SV,
-            ));
+            );
 
-            foreach ($data as $row) {
-                $this->namespaces[$row['prefix']] = $row['uri'];
-            }
+            $this->namespaces+= $namespaces;
 
             $this->originalNamespaces = $this->namespaces;
```

Comment:
yeah, but I think that makes a dependency mess. the node processor shouldn't know about the client in an ideal world. The namespaces should be held in an object and there should be only one instance of it internally.

`getNamespaces` is the public API, if we use only `getNamespaceObject` internally that should work? (at least that was the original plan).